tests/DontTest/DontDeserialiseTest.php

```diff
@@ -34,12 +38,40 @@ public function nonDeserialisableObject() : array
     {
         return [
             [NonDeserialisable::class],
+            [NonDeserialisableImplementingSerializable::class],
             [DontDoIt::class],
         ];
     }
 
     public function testSerialisePreventionIsFinal() : void
     {
         self::assertTrue((new \ReflectionMethod(DontDeserialise::class, '__wakeup'))->isFinal());
+        self::assertTrue((new \ReflectionMethod(DontDeserialise::class, 'unserialize'))->isFinal());
+        self::assertTrue((new \ReflectionMethod(DontDeserialise::class, '__unserialize'))->isFinal());
+    }
+
+    public function testManuallyInvokingMethods() : void
+    {
+        $dont = new DontDoIt();
+
+        try {
+            $dont->__unserialize([]);
+            self::markAsFailed('This method must always throw');
+        } catch (NonDeserialisableObject $_) {
```

Comment:
This should be multiple test methods, using `$this->expectException(NonDeserialisableObject::class)` each